src/main/java/mate/academy/dao/impl/MovieDaoImpl.java

```diff
@@ -47,12 +47,58 @@
     @Override
     public List<Movie> getAll() {
         try (Session session = HibernateUtil.getSessionFactory().openSession()) {
-            CriteriaQuery<Movie> criteriaQuery = session.getCriteriaBuilder()
-                    .createQuery(Movie.class);
-            criteriaQuery.from(Movie.class);
-            return session.createQuery(criteriaQuery).getResultList();
+            return session.createQuery("from Movie").list();
+        }
+    }
+
+    @Override
+    public Boolean update(Movie movie) {
+        SessionFactory sessionFactory = HibernateUtil.getSessionFactory();
+        Session session = null;
+        Transaction transaction = null;
+
+        try {
+            session = sessionFactory.openSession();
+            transaction = session.beginTransaction();
+            session.update(movie);
+            transaction.commit();
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("updating " + movie + " failed", e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+
+        return get(movie.getId()).get().equals(movie);
```

Comment:
The return statement in the `update` method retrieves the object from the database again to check if the update was successful. This can be inefficient. Consider returning `true` after a successful commit, assuming the transaction was successful.